grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -525,7 +525,7 @@ class StockMovementService {
                 substitutionItem.productId = associatedProduct.id
                 substitutionItem.productName = associatedProduct.name
                 substitutionItem.productCode = associatedProduct.productCode
-                substitutionItem.availableItems = availableItems
+                substitutionItem.availableItems = availableItems.findAll { it.inventoryItem.product.active }
```

Comment:
I think you should change the getProductAssociations (it's used in line 516) method to find only active products, instead of changing the available items